micro_ros_setup/README.md

```diff
@@ -1,80 +1,118 @@
+  - [Configuring micro-ROS firmware](#configuring-micro-ros-firmware)
+  - [Building micro-ROS firmware](#building-micro-ros-firmware)
+  - [Flashing micro-ROS firmware](#flashing-micro-ros-firmware)
+- [Building micro-ROS Agent](#building-micro-ros-agent)
+- [Contributing](#contributing)
 
-This version of the package targets  **ROS 2 Crystal** and has been tested on **Ubuntu 18.04** only! Other Linux distributions may work -- if you find issues, [please report them](https://github.com/micro-ROS/micro-ros-build/issues). 
+# Supported platforms
 
-For installing ROS 2, we recommend the [binary build of ROS 2](https://index.ros.org/doc/ros2/Installation/Crystal/). Note that if you use a non-package install, you may have to adjust some of the "source" commands below.
+| RTOS | Platform |  |
+|-|-|-|
+| [FreeRTOS](https://www.freertos.org/) | [Crazyflie 2.1](https://www.bitcraze.io/crazyflie-2-1/) | `freertos crazyflie21` | 
+| [Nuttx](https://nuttx.org/) | *Generic* ** | `nuttx` | 
+| Linux / Windows | *Host* * |
 
-We will use the standard ROS 2 meta build-tool, `colcon`, to compile. If you haven't used colcon, yet, you can probably just copy the instructions below as-is. I recommend [getting familiar with colcon](https://index.ros.org//doc/ros2/Tutorials/Colcon-Tutorial/), though, it has many useful options.
+*\* Support for compiling apps in native host for testing and debugging*
 
-## Concepts
+**\* Configuration for different platforms through configure_firmware.sh*
 
-Micro-ROS is a client-server system: The embedded micro-controller runs the client, and the "agent" runs on Linux (or potentially also Windows, though this package doesn't support that, yet).
+# Dependencies
 
-Software on a microcontroller is usually called "firmware" and it is cross-compiled on the host and then flashed onto the microcontroller. The following structure image shows the individual parts:
-![structure image](doc/structure.png)
+This package targets **ROS 2 Dashing** installation. Some other prerequisites needed for building a firmware using this package are:
 
-Correspondingly, in the micro-ROS build, we distinguish the firmware and the "regular" workspace for the server side. Since the regular workspace contains the agent, in other documentation it also sometimes called "agent workspace".
-
-# Installation
-
-This is, overall, a four-step process.
+```
+sudo apt install python-rosdep
+```
 
-## 1/4 Setting up the workspace
+# Building 
 
-It is best to start with a fresh workspace, both to speed up compiles and to avoid unintended interference. 
+Create a ROS 2 workspace and build this package:
 
-```shell
+```
 source /opt/ros/dashing/setup.bash
-sudo apt install python-rosdep
-mkdir -p uros_ws/src
+mkdir uros_ws
 cd uros_ws
-git clone --recursive https://github.com/micro-ROS/micro-ros-build.git src/micro-ros-build
+git clone --recursive -b feature/multiboard https://github.com/micro-ROS/micro-ros-build.git src/micro-ros-build
```

Comment:
Please update it taking into account the target branch.

Reply:
Err... Once we merge this, that will no longer be necessary, because it's going to be on the dashing branch!

Reply:
Done

Reply:
I think it could be necessary for future versions... Maybe we can change that by
```
git clone --recursive -b <version=|crystal??|dashing|foxie|> https://github.com/micro-ROS/micro-ros-build.git src/micro-ros-build
```
What do you think @iluetkeb ?
Sorry because I've merged before I see your comments :-(

Reply:
Ah, right. If we're checking out something other than the default, it is necessary, so we should put it in there. But *not* feature/multiboard, since that's going to be deleted soon, but rather "dashing".

Do we need recursive checkout? I think not.

Reply:
Ingo, please check this PR: https://github.com/micro-ROS/micro-ros-build/pull/43

If it match your requirements please merge squash and delete my branch.

Thanks!